fix: ci-network-scenario label, fix nightly benchmarks

ludamad opened the pull request:
## Summary
- Split out the handling of per-release scenario tests to be part of ci3.yml. Simplify test-network-scenarios.yml for the workflow_dispatch case. 
- We now support the label ci-network-scenarios. You may combine this with ci-skip to not also run normal tests.
- Nightly spartan benchmarks flow yaml working and simplified. 
- Work on the redact script, which now handles github's add-mask (needed to use our ci3 logging utils on the parts of the sript)

## Fixes
- Allow for overriding namespace in env files
- Only pass ETHERSCAN_API_KEY to terraform when VERIFY_CONTRACTS is true
- Refactoring for YAML's that call into ci
- Use nested logging and k8s URL links for much better deploy_network.sh observability

Example workflow_dispatch passing for test-network-scenarios.yml
https://github.com/AztecProtocol/aztec-packages/actions/runs/20379119834/job/58564980277

Example ci-network-scenario label passing: 
https://github.com/AztecProtocol/aztec-packages/actions/runs/20385820878/job/58586990753?pr=19140 (lost track of the one later where teardown was fixed)

just-mitch commented on .github/workflows/ci3.yml:
```diff
@@ -19,14 +19,30 @@ concurrency:
+  #
+  #            Standard PR)                      Merge Queue
+  #                  |                                 |
+  #      |  1x AMD64 EC2 Spot  |           |  [ 4x ARM64 EC2 Spot ]  |
```
```suggestion
  #      |  1x AMD64 EC2 Spot  |           |  [ 4x AMD64 EC2 Spot ]  |
```
Right?

just-mitch commented on spartan/scripts/network_teardown.sh:
```diff
@@ -0,0 +1,26 @@
+
+# Delete Chaos Mesh experiments if they exist
+echo "Cleaning up Chaos Mesh experiments in namespace $NAMESPACE..."
+kubectl delete podchaos,networkchaos,stresschaos,iochaos,timechaos,kernelchaos,dnschaos,httpchaos,blockchaos,jvmchaos,physicalmachinechaos \
```
Agreed. It is surprisingly difficult to kill chaos(mesh).

just-mitch commented on spartan/scripts/setup_gcp_secrets.sh:
```diff
@@ -94,59 +60,146 @@ declare -A SECRET_MAPPINGS=(
 )
 
-# Replace placeholders with actual secrets
+# --- Preparation ---
```
I was getting a bit nervous about this setup, so I searched it up and found https://github.com/google-github-actions/get-secretmanager-secrets ? Alternatively, AI suggested:

1. Create .env.template:
```
# Instead of "REPLACE_WITH_...", use standard shell variable syntax
ETHEREUM_RPC_URLS=${ETHEREUM_RPC_URLS} # Suffixes work natively in bash/envsubst
SOME_URL=${BASE_SECRET_URL}/v1/api
```
2. Use envsubst (part of gettext package): This tool replaces environment variables in text streams.
```
#!/bin/bash
# 1. Fetch secrets into MEMORY (Environment Variables)
# Using 'export' prevents writing to disk
export ETHEREUM_RPC_URLS=$(gcloud secrets versions access latest --secret="..." ...)
export BASE_SECRET_URL=$(gcloud secrets versions access latest --secret="..." ...)
# 2. Render the file
# envsubst reads the template, fills in the variables from memory, writes the file
envsubst < .env.template > .env
```

ludamad replied:
I can revert this

just-mitch replied:
I think it is better than what we have, I just wanted to surface other ideas 👍

just-mitch commented on ci3/redact:
```diff
@@ -1,29 +1,101 @@
+#   | Version  | Time  | Speedup    |
+#   |----------|-------|------------|
+#   | Bash/awk | 9.84s | -          |
+#   | Perl     | 0.41s | 24x faster |
```
🔥

just-mitch reviewed (approved):
Really nice. I think the secrets stuff is getting a little out of hand though. Don't need to change in this PR.

ludamad committed:
fix: ci-network-scenario label, fix nightly benchmarks

## Summary
- Split out the handling of per-release scenario tests to be part of ci3.yml. Simplify test-network-scenarios.yml for the workflow_dispatch case.
- We now support the label ci-network-scenarios. You may combine this with ci-skip to not also run normal tests.
- Nightly spartan benchmarks flow yaml working and simplified.
- Work on the redact script, which now handles github's add-mask (needed to use our ci3 logging utils on the parts of the sript)

## Fixes
- Allow for overriding namespace in env files
- Only pass ETHERSCAN_API_KEY to terraform when VERIFY_CONTRACTS is true
- Refactoring for YAML's that call into ci
- Use nested logging and k8s URL links for much better deploy_network.sh observability

Example workflow_dispatch passing for test-network-scenarios.yml
https://github.com/AztecProtocol/aztec-packages/actions/runs/20379119834/job/58564980277

Example ci-network-scenario label passing:
https://github.com/AztecProtocol/aztec-packages/actions/runs/20385820878/job/58586990753?pr=19140 (lost track of the one later where teardown was fixed)
